common/src/Utilities/FileUtils.cpp

```diff
@@ -0,0 +1,139 @@
+/*  PCSX2 - PS2 Emulator for PCs
+ *  Copyright (C) 2002-2021  PCSX2 Dev Team
+ *
+ *  PCSX2 is free software: you can redistribute it and/or modify it under the terms
+ *  of the GNU Lesser General Public License as published by the Free Software Found-
+ *  ation, either version 3 of the License, or (at your option) any later version.
+ *
+ *  PCSX2 is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PrecompiledHeader.h"
+#include "FileUtils.h"
+#include <wx/file.h>
+#include <wx/stdpaths.h>
+#include <wx/filename.h>
+#include "ghc/filesystem.h"
+#include <wx/utils.h>
+
+// ---------------------------------------------------------------------------------
+//  Path namespace
+// ---------------------------------------------------------------------------------
+bool FileUtils::IsRelative(const fs::path &path)
+{
+    return fs::path(path).is_relative();
+}
+
+bool FileUtils::IsDirectoryWithinDirectory(fs::path base, fs::path dir)
+{
+    fs::path relativePath = fs::relative(fs::absolute(dir), fs::absolute(base));
+    if (relativePath.empty())
+    {
+        return false;
+    }
+    return relativePath.string().rfind("..", 0) != 0;
+}
+
+s64 FileUtils::GetFileSize(const fs::path &path)
+{
+    if (!fs::exists(path))
+        return -1;
+    return (s64)fs::file_size(path);
+}
+
+fs::path FileUtils::Normalize(const fs::path &src)
+{
+    if (src.empty())
+    {
+        return fs::path();
+    }
+    return src.lexically_normal();
+}
+
+fs::path FileUtils::MakeAbsolute(const fs::path &src)
+{
+    return ghc::filesystem::absolute(src);
+}
+
+fs::path FileUtils::Combine(const fs::path &srcPath, const fs::path &srcFile)
+{
+    return (srcPath / srcFile).make_preferred();
```

Comment:
I'm still fairly sure the `make_preferred` is completely unnecessary, and we should just use the `/` operator